SIT.Manager.Avalonia/ManagedProcess/ManagedProcess.cs

```diff
@@ -48,10 +48,12 @@ public virtual void Stop()
 
         // Stop the server process
         bool clsMsgSent = _process.CloseMainWindow();
-        if (!clsMsgSent)
-            _process.Kill();
-
-        _process.WaitForExit();
-        _process.Close();
+        if (clsMsgSent)
+        {
+            if (!_process.WaitForExit(TimeSpan.FromSeconds(5)))
+            {
+                _process.Kill();
+            }
+        }
```

Comment:
'clsMsgSent' should be always false if close after launch isn't on though is it not meaning this will never actually run?

Would something like this work better?

```csharp
        bool closed = false;

        // Stop the server process
        bool clsMsgSent = _process.CloseMainWindow();
        if (clsMsgSent)
        {
            closed = _process.WaitForExit(TimeSpan.FromSeconds(5));
        }

        if (!closed)
        {
            _process.Kill();
            _process.WaitForExit(TimeSpan.FromSeconds(5));
        }
 
        _process.Close();

```

Reply:
You are absolutely correct. Writing code while sleep deprived gives interesting results